File: isAnagram.py
# This may be a lot of code, but I am currently re learning python and do not know the available python methods quite yet.

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        hashMap1 = {}
        hashMap2 = {}

        for char in s:
            if(char not in hashMap1):
                hashMap1[char] = 1
            else:
                hashMap1[char] = hashMap1[char]+1

        for char in t:
            if(char not in hashMap2):
                hashMap2[char] = 1
            else:
                hashMap2[char] = hashMap2[char]+1   

        if len(hashMap1) != len(hashMap2):
            return False

        for key in hashMap1:
            if key not in hashMap2 or hashMap1[key] != hashMap2[key]:
                return False
        
        return True

File: isAnagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> hashMap1 = new HashMap<>();
        HashMap<Character, Integer> hashMap2 = new HashMap<>();

        if(s.length()!=t.length()) return false;

        for(int i = 0 ; i < s.length() ; i ++){
            char a = s.charAt(i); char b = t.charAt(i);

            hashMap1.putIfAbsent(a, 0);
            hashMap2.putIfAbsent(b, 0);

            hashMap1.put(a, hashMap1.get(a)+1);
            hashMap2.put(b, hashMap2.get(b)+1);
        }

        if(hashMap1.size()!=hashMap2.size()) return false;

        for(char x : hashMap1.keySet()){
            if(hashMap2.get(x)==null || !hashMap1.get(x).equals(hashMap2.get(x)))
                return false;
        }
        
        return true;
    }
}
